### ISSUE TITLE
RAM Usage

### ISSUE THREAD

**Issue by antono2** [v-analyzer/v-analyzer#60]
### Describe the bug

While working on a big V file my pc suddenly froze.
After restarting and another few hours, I checked what's slowing everything down again.
The v-analyzer was taking up 1.6gb , each instance.

Using the C intermediary format we can compare the same code with the C language server ccls.
The V-analyzer ram usage keeps growing, while ccls stays around 50mb for me, meaning v-analyzer doesn't seem to free memory.

### Expected Behavior

Use around 50mb of RAM, like ccls, for the same code in form of the C intermediary.

### Current Behavior

Memory is never freed.
![Screenshot from 2023-12-07 07-00-46](https://github.com/v-analyzer/v-analyzer/assets/28753789/525bc867-c69b-49fc-9a4f-8ee6d07890be)

### Reproduction Steps

Find some bigger V file, e.g. translate  the vulkan header files:

1. Download vulkan sdk `https://vulkan.lunarg.com/sdk/home`
2. `v translate 1.3.268.0/x86_64/include/vulkan/vulkan.h`
3. Use v-analyzer with `vulkan.v` and go to definition or find references a few times

### Possible Solution

You tell me please.
What did you do to investigate?

### Environment details (`v doctor` output)

V full version: V 0.4.3 46086c0.21d4407
OS: linux, Ubuntu 22.04.3 LTS
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz

getwd: /home/anton
vexe: /home/anton/workspace/v/v
vexe mtime: 2023-12-05 21:29:19

vroot: OK, value: /home/anton/workspace/v
VMODULES: OK, value: /home/anton/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.34.1
Git vroot status: weekly.2023.30-933-g21d44072 (4 commit(s) behind V master)
.git/config present: true

CC version: cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

### Editor name

kakoune

### v-analyzer Version

v-analyzer version 0.0.2-beta.1

**Comment by antono2** [comment]
I could change the memory growth rate, but it still grows without limit I think.
Change the v command in build.vsh and add the garbage collector. Feel free to experiment with the other `-gc` options.
`pub const base_build_command = '${@VEXE} ${code_path} -gc boehm -o ${bin_path} -no-parallel'
`

Also install Boehm-Demers-Weiser's GC
`sudo apt install libgc-dev`

Then you can rebuild as usual
`cd v-analyzer; v build.vsh release`

The result is 500MB RAM usage instead of 1.8GB.

**Comment by JalonSolov** [comment]
`-gc boehm` is the default for V (for a while, now), and you do not need to install libgc separately - V comes with it already, in the `thirdparty/libgc` subdir.  If you install it using `apt`, that version won't be used by V.

**Comment by antono2** [comment]
You are right @JalonSolov , the boehm gc doesn't have any effect. Guess I used find references on a different symbol.
@ttytm Unfortunately the issue still persists with the latest updates.
After some find references:
![v-analyzer_log](https://github.com/v-analyzer/v-analyzer/assets/28753789/f6471b86-0dce-49b3-986b-dc26b72d5107)

![v-analyzer boehm](https://github.com/v-analyzer/v-analyzer/assets/28753789/60a4ccdb-fc6a-4939-9768-33ccb6c3bbc7)

**Comment by ttytm** [comment]
In which editor are you using v-analyzer? Might the memory usage be due to many spawned processes that were not ended when they should have been ended? The screenshots look like they could be cut off.

**Comment by antono2** [comment]
@ttytm kakoune. The screenshots are cut off at the last instance of v-analyzer. I only edited 1 file.
Spawned processes are ended by kakoune based on a config setting. But yes there are 2 v-analyzer spawned, one of which stays around for that time set per config. That one is no the problem though, as it always uses 145.1mb.
